src/components/structures/MatrixChat.tsx

```diff
@@ -1900,10 +1900,12 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
             return setLoggedInPromise;
         }
 
-        // Test for the master cross-signing key in SSSS as a quick proxy for
-        // whether cross-signing has been set up on the account.
-        const masterKeyInStorage = !!cli.getAccountData("m.cross_signing.master");
-        if (masterKeyInStorage) {
+        // wait for the client to finish downloading cross-signing keys for us so we
+        // know whether or not we have keys set up on this account
+        await cli.downloadKeys([cli.getUserId()]);
```

Comment:
do these not come down the initial sync we awaited earlier? If not then please parallelize so it can be done whilst waiting for the initialSync.

Reply:
It turns out this was a bit of a mess, as hopefully explained in the commit comment. Should be better now hopefully and will download keys in parallel.